packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -23,6 +23,26 @@ import Header from '../Header/Header'
 import Portal from '../Portal/Portal'
 import Flex from '../Flex/Flex'
 
+const getOrGenerateIdFromShorthand = (
```

Comment:
👍 probably, in future we will need to move this function to `lib`, as the logic of `peek if client had specified some prop for shorthand value` seems to be a quite common case

Reply:
may just suggest to cover React element's case for shorthand slot in tests

Reply:
Was done in https://github.com/stardust-ui/react/pull/1449/commits/e816e3c8f5cc45f1cb62cea070c93ea9d7dcdd9c